client/state/data-layer/wpcom/comments/index.js

```diff
@@ -58,6 +117,16 @@ export const addComments = ( { dispatch }, { siteId, postId }, next, { comments,
 	}
 };
 
+export const writePostCommentSuccess = ( { dispatch, getState }, { siteId, postId, parentCommentId, placeholderId }, next, comment ) => {
+	// remove placeholder from state
+	dispatch( { type: COMMENTS_REMOVE, siteId, postId, commentId: placeholderId } );
+	// add new comment to state with updated values from server
+	dispatch( { type: COMMENTS_RECEIVE, siteId, postId, comments: [ comment ], skipSort: !! parentCommentId } );
+	// increment comments count
+	const commentCount = getPostTotalCommentsCount( getState(), siteId, postId );
+	dispatch( { type: COMMENTS_COUNT_RECEIVE, siteId, postId, totalCommentsCount: commentCount + 1 } );
```

Comment:
The network request is something I think is good to keep. Two points on this…

 - if we wanted to increment the comment count automatically then we should probably let the `commentCount` reducer do it for us. We are cross-cutting concerns with this code and that will make it harder to understand why it changed.

 - it's likely that other comments are being created simultaneously here. incrementing the count is probably getting us closer to the total, but we'll have to update anyway from the network if we want an accurate new count. it seems good to me to continue.

the use of `_RECEIVE` as an action here I think is showing its darker side. when we have `ADD_COMMENT` and maybe `UPDATE_COMMENT` action types we should be left with less ambiguity that makes these tightly-coupled things clearer.